src/components/icon.scss

```diff
@@ -0,0 +1,25 @@
+// * Icon component
+// *
+// * 1. Fine tune the positioning of the icon
+// * 2. Fill with currentColor by default
+// *
+
+.Icon {
+  position: relative; // 1
+  margin: 0;
+  display: inline-block;
+  width: 1em;
+  height: 1em;
+  font-size: 1em;
+  vertical-align: middle;
+  top: -1px; // 1
+  fill: currentColor; // 2
+}
+
+// * Modifier -- large icon relative to parent text
+
+$Icon--large-size: 1.25em;
+
+.Icon--large {
```

Comment:
I think if we have a font-size increment ratio that we are using, this becomes easier. Then we can have font sizing utilities that increase or decrease font size, and can be used on icons too. It sounds like you are working on some typographic scale stuff, so maybe we wait until then to implement any icon sizing. I know in the past it's been very useful to have a way to size icons differently in different situations, without having to rely on a component for this. We can see what we need once we start actually using them. I'll remove for now.